src/concrete_solve.jl

```diff
@@ -211,7 +211,10 @@ function DiffEqBase._concrete_solve_adjoint(prob,alg,sensealg::AbstractForwardSe
        end
        J'v
      end
-     (nothing,nothing,nothing,adj,nothing,ntuple(_->nothing, length(args))...)
+     du0 = ChainRulesCore.@thunk begin
+         error("ForwardSensitivity does not differentiate with respect to u0. Change your sensealg.")
+     end

```

Comment:
```suggestion
     du0 = ChainRulesCore.@not_implemented(
         "ForwardSensitivity does not differentiate with respect to u0. Change your sensealg."
     )
```
https://juliadiff.org/ChainRulesCore.jl/dev/api.html#ChainRulesCore.@not_implemented-Tuple{Any}

@devmotion  added `@not_implemented` returning it's own differential type.
Because if an AD system like Zygote eagerly `unthunk`s everything, then even if this output is not used  it would still error.